src/app/my-rewards/components/AllTimeShare.tsx

```diff
@@ -0,0 +1,44 @@
+import { RequiredBuilder } from '@/app/collective-rewards/types'
+import { useGetBuildersByState } from '@/app/collective-rewards/user'
+import { useHandleErrors } from '@/app/collective-rewards/utils'
+import { Span } from '@/components/TypographyNew/'
+import { getTokens } from '@/lib/tokens'
+import { Address } from 'viem'
+import { useBuilderAllTimeShare } from '../hooks/useBuilderAllTimeShare'
+import { RewardCard } from './RewardCard'
+
+export interface RewardCardAllTimeShareProps {
+  isLoading: boolean
+  amount: string
+}
+
+export const AllTimeShare = ({ gauge }: { gauge: Address }) => {
+  const { data: activatedBuilders, error: activatedBuildersError } = useGetBuildersByState<RequiredBuilder>({
+    activated: true,
+  })
+  const activatedGauges = activatedBuilders?.map(({ gauge }) => gauge) ?? []
+
+  useHandleErrors({ error: activatedBuildersError, title: 'Error loading gauges' })
+
+  const rifAddress = getTokens().rif.address
```

Comment:
Are we going to use this approach of the function or change it to a constant?

Reply:
We're using the constant